src/gamepad/gamepad-monitor.vala

```diff
@@ -0,0 +1,78 @@
+// This file is part of GNOME Games. License: GPLv3
+
+/**
+ * This class provides a way to the client to monitor gamepads
+ *
+ * The client interfaces with this class primarily
+ */
+public class LibGamepad.GamepadMonitor : Object {
+	/**
+	 * Emitted when a gamepad is plugged in
+	 * @param  gamepad    The gamepad
+	 */
+	public signal void gamepad_plugged (Gamepad gamepad);
+
+	/**
+	 * Emitted when a gamepad is unplugged
+	 * @param  gamepad    The gamepad
+	 */
+	public signal void gamepad_unplugged (Gamepad gamepad);
+
+	public delegate void GamepadCallback (Gamepad gamepad);
+
+	private static HashTable<string, RawGamepad> identifier_to_raw_gamepad;
+	private static HashTable<string, string> guid_to_raw_name;
+
+	private RawGamepadMonitor raw_gamepad_monitor;
+
+	public GamepadMonitor () {
+		init_static_if_not ();
+
+		raw_gamepad_monitor = new LinuxRawGamepadMonitor ();
+
+		raw_gamepad_monitor.gamepad_plugged.connect (on_raw_gamepad_plugged);
+		raw_gamepad_monitor.gamepad_unplugged.connect (on_raw_gamepad_unplugged);
```

Comment:
Let's instead monitor the `unplugged` signal of the created gamepads: when a gamepad is unplugged, we can directly remove it.